### ISSUE TITLE
Czech translation

### ISSUE THREAD

**Issue by Flippo008** [axelboc/anki-ultimate-geography#339]
Hi, 
I read #313 and my case is totally the same. I have been learning Ultimate geography and making my own Czech translation because for me is crucial to know the names of countries etc. in my language. After v4.0 release I am planning to make a clean import and start from a scratch since the translation wasn't as good as I would like to be for public sharing. Same as hoi3 I don't know how to proceed correctly and I also read your answer that the maximum number of translation was reached. But I will make the Czech translation for my purpose regardless of sharing the translation to other people, so if the work could be incorporated later in the deck, that would be just a sweet bonus for me.

Sorry for eventual mistakes in this issue, obviously I am not a native speaker. 

I am looking forward to v4.0 of Ultimate geography.

**Comment by axelboc** [comment]
This is so awesome! 😁 I'm really excited to see so many of you wanting to contribute a translation.

@Flippo008, @emiham, @pt-pt-guy, @hoi3, @GrimPixel, now that the inclusion rules have been applied and that v4 is (almost) out, I think you can safely start your translations. Here is the process I recommend, which should allow us to integrate your translations into the deck fairly easily, once the new deck manager is in place and #143 is resolved:

1. Fork this repo and create a branch for your translation.
1. Open `data.csv` in a spreadsheet editor and remove the columns of all the translations, leaving only the original English fields. You can also remove the `Flag`, `Map` and `Tags` columns, but make sure to keep the `guid` column.
1. Add columns for your own translations, following the correct naming convention (e.g. `Country info:de`, where `de` is the [language code](https://r12a.github.io/app-subtags/)).
1. Fill in the blanks, and commit the changes to `data.csv` to your branch as you go!
1. Keep an eye on this repository for changes to the English content, and try to integrate them as you go.

---

If you prefer translating the deck in Anki, that's fine too. There are some pitfalls, like not being able to upgrade the deck while you're translating it, but I doubt it will be a problem, since there's little chance we'll make a release before the new deck manager comes along. You never know, though... 😄

If you go the Anki route, once we're ready to integrate your translation, you'll need to [set up a fresh repo with Anki DM](https://github.com/OnkelTem/anki-dm#installation), export your deck with CrowAnki, and [get Anki DM to disassemble it](https://github.com/OnkelTem/anki-dm#usage). The GUIDs won't be the same, but hopefully we can use one of the common fields like the `Flag` field to match the notes.

---

Good luck!! 💪

**Comment by aplaice** [comment]
I had been planning to make a comment along the same lines! :)

<hr/>

> Open data.csv in a spreadsheet editor and remove the columns of all the translations, leaving only the original English fields. You can also remove the Flag, Map and Tags columns, but make sure to keep the guid column.

Should we create such a "dummy" `data.csv`, say in a branch, to possibly avoid confusion (?) and duplicated work? (i.e. have a `data.csv` with only the English/shared columns and `:xx` columns?) (I could do that quickly, if yes.)

<hr/>

> The GUIDs won't be the same

If they'll be translating the English deck in-place (I can't really think of any other options), then why would the GUIDs be different?

<hr/>

Do we know how the people who made the previous translations worked?

**Comment by axelboc** [comment]
> Should we create such a "dummy" data.csv, say in a branch, to possibly avoid confusion (?) and duplicated work? (i.e. have a data.csv with only the English/shared columns and :xx columns?) (I could do that quickly, if yes.)

Great idea, thanks!

> If they'll be translating the English deck in-place (I can't really think of any other options), then why would the GUIDs be different?

Because I think Anki DM creates new GUIDs when disassembling the deck. I could be wrong, though.

> Edit: Should we continue the general translation discussion in a new issue?

See #340. I've replaced the steps with your recommendation. I used the name `translation` for the branch.

**Comment by aplaice** [comment]
(Replying to https://github.com/axelboc/anki-ultimate-geography/issues/340#issuecomment-657089762):

> Sorry for the late response, I was a couple of days off PC.

No problem!

> I copied my data to data.csv in translation branch, so if I did it right, you should be able to check my translation.

Thanks very much! I understand that the translation is [here](https://github.com/Flippo008/anki-ultimate-geography/tree/patch-1)?

There seems to be an encoding issue (e.g. there's `AfghĂˇnistĂˇn` instead of `Afghánistán` and so on). It almost looks like Windows-1250 (?) encoded as UTF-8, but not quite. However, I'm not sure what caused it. :( (It might have been `anki-dm` or `CrowdAnki`'s interplay with the system locale or even a copy-paste encoding issue.) Could you perhaps upload your CrowdAnki `deck.json` as a GitHub gist?

A second issue is that there are unnecessary HTML tags `<div>` and `</div>` inserted into the fields, which is definitely Anki's fault (it adds them when you insert a new line, and keeps them there, even when they're no longer needed). However, they could be straightforwardly removed once this is ready for merging, so don't worry about them.

**Comment by Flippo008** [comment]
I am back. I apologize for not responding, I was on summer camp, so I couldn't really upload the wanted file. Sorry. Here is URL for `deck.json` uploaded as GitHub gist.

https://gist.github.com/Flippo008/2c59d9abc4c273a6e7a84c3a74cfc729

**Comment by aplaice** [comment]
Thanks very much!

`deck.json` doesn't have any encoding issues, so I guess that the problem had been caused by PHP during `anki-dm`'s conversion of `deck.json` into `data.csv`. Since we'll be switching away from `anki-dm` it's not worth delving deeper.

On a cursory look through the translation (by a non-Czech speaker), it looks great! :)

**Comment by ohare93** [comment]
New deck manager is pulled :+1: The source csv has now been broken into multiple separate csv files inside `src/data`. 

You can now add your translation :clap: This requires 2 steps to do it yourself:
1) Add the translation as a column in each of the source files.
2) Edit the Brain Brew recipe in `recipes/source_to_anki.yaml` to build your new column to your new language deck. Just copy pasting what the others do, and changing it to match your new columns will do :+1: 

I am happy to do the second step, if you do the first :wink: 

Better yet, if you have created the translation inside Anki itself then I can do it all in <5mins, as the new deck manager can also import _from CrowdAnki_ to our Source control. Feel free to export the deck, upload a zip of it here, and I will do magic :clap: :grin:

**Comment by aplaice** [comment]
> Better yet, if you have created the translation inside Anki itself then I can do it all in <5mins, as the new deck manager can also import from CrowdAnki to our Source control. Feel free to export the deck, upload a zip of it here, and I will do magic clap grin

I think that @Flippo008 has already produced a `deck.json` [above](https://github.com/axelboc/anki-ultimate-geography/issues/339#issuecomment-665549681), so apart from changes to the deck since July, the translation is mostly ready!

**Comment by aplaice** [comment]
@Flippo008, if you're still willing to continue with your translation, (no pressure! and sorry for the delay!), then all that we'd need to finish would be for a couple of translations of the minor changes and additions, since then, if/when you have time.

AFAICT your translation was based on the stable v4.0 release.  The following relevant content changes have been made since v4.0:

1. Change of the name of the "continent"/"part of the world" Australia to Oceania, and an addition of a "Country info":

    ```
    World region covering the Australian continent and most of the islands in the Pacific Ocean.
    ```

    We decided that it made more sense in the context of the deck to have Oceania rather than Australia, especially since that's what the map shows. (I think that the distinction between Australia and Oceania is the same in English as in Czech.)

2. Addition of 7 new notes (with only Map → Country cards, and hence only needing translation of the Country name and the Country info):

| Country | Country info |
|-----------|-----------------|
|Wallis and Futuna|Overseas territory of France.|
|Akrotiri and Dhekelia|Overseas territory of the United Kingdom.|
|Svalbard|Unincorporated internal area of Norway.|
|Hawaii|State of the United States.|
|Alaska|State of the United States.|
|Bougainville|Autonomous region of Papua New Guinea.|
|Jeju|Autonomous province of South Korea.|

Overall, I think that it'd be easiest if you just give the translations of the 7 new "Country" names and  8 "Country infos" (Oceania and the 7 new ones) here, which I (or @ohare93) could add on top of your CrowdAnki deck from above — otherwise you and we might have issues with overwriting of changes/GUIDs etc. (At the end of the process you'll be able to update your current Anki deck, with the updated version (including new notes!), as translated by you, without loss of your current translation/notes, and you'll also be able to easily continue updating the deck in the future.)

**Comment by Flippo008** [comment]
Glad to hear this good news. Yes, you are correct that my translation was based on v4.0 release. So if I understand correctly, I write these changed/added information here and you will edit it. So:

Austrálie (Oceánie) -> Oceánie
Country info: Světová oblast zahrnující australský kontinent a většinu ostrovů v Tichém oceánu. 

here is a table with a corresponding order
| Country  | Country info |
| ------------- | ------------- |
| Wallis a Futuna  | Zámořské území Francie.  |
| Akrotiri a Dekelia  | Zámořské území Velké Británie.  |
| Špicberky  | Nezařazené území Norska.  |
| Havaj  | Stát spojených států.  |
| Aljaška  | Stát spojených států.  |
| Bougainville  | Autonomní území státu Papua Nová Guinea.  |
| Čedžu  | Autonomní provincie Jižní Koreji  |

I know we are respecting title of Wikipedia page and as it seems when it is obvious that info is redundant, it gets deleted. 
e.g. Autonomous Region of Bougainville -> Bougainville 

And id I understood correctly, in the future I will have access to a file (like data.cvs) where I can edit new deck updates or some minor translation mistakes.

**Comment by aplaice** [comment]
@Flippo008 Thanks for the very fast reply and for the translations!

> So if I understand correctly, I write these changed/added information here and you will edit it.

Yes. (Otherwise creating an appropriate `deck.json` with correct GUIDs could have been messy.)

> I know we are respecting title of Wikipedia page and as it seems when it is obvious that info is redundant, it gets deleted.
e.g. Autonomous Region of Bougainville -> Bougainville

Yes, exactly, like you've translated.

> And id I understood correctly, in the future I will have access to a file (like data.cvs) where I can edit new deck updates or some minor translation mistakes.

Yes! It's now split between several files, but the idea is exactly the same.

<hr/>

@ohare93 can I do the magic, given that I had volunteered slightly more than you had volunteered? Or do you want to do the magic and I'll add the several manual translations to the CSVs, as a pull request on top of that?

**Comment by ohare93** [comment]
> @ohare93 can I do the magic, given that I had volunteered slightly more than you had volunteered? Or do you want to do the magic and I'll add the several manual translations to the CSVs, as a pull request on top of that?

Yes please! :grin: :handshake:

**Comment by aplaice** [comment]
@ohare93, (I think) I've done some of the magic and imported the Czech translation, but the sorting of all the CSV files is reset to alphabetical. Do you have any tips for getting around this?

**Comment by ohare93** [comment]
> @ohare93, (I think) I've done some of the magic and imported the Czech translation, but the sorting of all the CSV files is reset to alphabetical. Do you have any tips for getting around this?

Pull (or just redo) my changes to the recipe files [here](https://github.com/axelboc/anki-ultimate-geography/compare/master...ohare93:RussianTranslationInBrainBrew?expand=1) I just faced a similar annoyance doing the Russian :sweat_smile:

**Comment by ohare93** [comment]
Taking out the sorting will only make it not sort in the future, it will not undo the sorting it has already done. So you may have to rollback your changes to the split csvs and run the recipe again. Hopefully that won't mean you have to redo too much! :sweat: 

I have included removing the sorting in the Russian translation PR, so this should not happen again.

**Comment by aplaice** [comment]
That worked, thanks! :)
